src/Native/LdaNative/alias_multinomial_rng_int.hpp

```diff
@@ -108,7 +110,8 @@ namespace wood
             int32_t H_head = 0;
             int32_t H_tail = 0;
 
-            for (auto i = 0; i < proportion_int_.size(); ++i)
+			int32_t size = static_cast<int32_t>(proportion_int_.size());
```

Comment:
Isn't it more correct to use `std::vector<int>::size_type` here?

```
for(std::vector<int>::size_type i = 0; i != v.size(); i++) {
    /* std::cout << v[i]; ... */
}
```

Reply:
yes, will fix

---
In reply to: [276748210](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276748210) [](ancestors = 276748210)

Reply:
I agree - it is more correct, but you would have to cast when setting L_[L_tail].first and H_[H_tail].first to int32_t.

Gleb - should this fail if the size is bigger than int32_t max? If so, it may warrant a check first then cast the size as this cast will not throw during runtime. Also the cast can change to size=(int32_t)proportion_int.size();


---
In reply to: [276748210](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276748210) [](ancestors = 276748210)

Reply:
I think I fixed it. Please check.

---
In reply to: [276826843](https://github.com/dotnet/machinelearning/pull/3370#discussion_r276826843) [](ancestors = 276826843,276748210)